Source/Core/Common/Matrix.cpp

```diff
@@ -78,7 +78,7 @@ Quaternion::Quaternion(float w, float x, float y, float z) : data(x, y, z, w)
 
 float Quaternion::Norm() const
 {
-  return data.Dot(data);
+  return std::sqrt(data.Dot(data));
 }
```

Comment:
[Reference](https://en.wikipedia.org/wiki/Quaternion#Conjugation,_the_norm,_and_reciprocal); note that this is computing `sqrt(a^2 + b^2 + c^2 + d^2)`. It looks like `Norm()` isn't used anywhere other than for `Normalized()` so a `Norm2` function probably isn't needed.